docs/src/modules/java-protobuf/pages/workflows.adoc

```diff
@@ -0,0 +1,200 @@
+= Implementing Workflows
+:page-aliases: java-protobuf:workflow.adoc
+:page-supergroup-java-scala: Language
+
+include::ROOT:partial$include.adoc[]
+
+Workflows make it possible to implement long-running, multi-step business processes while focusing exclusively on domain and business logic. Workflows provide durability, consistency and the ability to call other components and services. Business transactions can be modeled in one central place, and the Workflow will keep them running smoothly, or roll back if something goes wrong.
+
+
+
+
+include::java-protobuf:partial$effects-workflow.adoc[]
+
+See also xref:java:development-process.adoc#_effects_api[Understanding what an Effect is]
+
+== Defining the proto files
+
+We want to build a simple workflow that transfers funds between two wallets. Before that, we will create a wallet subdomain with some basic functionalities that we could use later. For simplicity a `WalletEntity` is implemented as a xref:value-entity.adoc[ValueEntity], but for a production-ready solution an xref:event-sourced-entities.adoc[EventSourcedEntity] would be a better choice.
+
+[source,java,indent=0]
+.src/main/proto/com/example/wallet/wallet_api.proto
+----
+include::example$java-protobuf-transfer-workflow/src/main/proto/com/example/wallet/wallet_api.proto[tag=wallet]
+----
+<1> An endpoint to create a wallet with an initial balance.
+<2> An endpoint to withdraw funds from the wallet.
+<3> An endpoint to deposit funds to the wallet.
+<4> An endpoint to get current wallet balance.
+
+Now we can focus on the workflow implementation itself. A workflow has state, which can be updated in command handlers and transition methods. During the state modeling we might consider the information that is required for validation, running the steps, collecting data from steps or tracking the workflow progress.
+
+[source,java,indent=0]
+.src/main/proto/com/example/transfer/transfer_domain.proto
+----
+include::example$java-protobuf-transfer-workflow/src/main/proto/com/example/transfer/transfer_domain.proto[tag=transfer-state]
+----
+<1> A `TransferState` encapsulates data required to withdraw and deposit funds.
+<2> A `TransferStatus` is used to track workflow progress.
+
+The following `transfer_api.proto` file defines our `TransferWorkflow` component. In this file we instruct the Kalix code generation tooling (codegen) to generate all stubs for your workflow/service and corresponding tests, as well as an abstract class for your implementation to extend.
+
+[source,java,indent=0]
+.src/main/proto/com/example/transfer/transfer_api.proto
+----
+include::example$java-protobuf-transfer-workflow/src/main/proto/com/example/transfer/transfer_api.proto[tag=transfer-workflow]
+----
+<1> Any classes generated from this protobuf file will be in the Java package `com.example.transfer.api`.
+<2> Import the Kalix protobuf annotations or options.
```

Comment:
This doesn't look correct referring to 
```
import "google/protobuf/empty.proto"; 
import "kalix/annotations.proto";
import "com/example/transfer/transfer_domain.proto"
```

Reply:
why? because we import other things as well?